### PR TITLE
Enhance a bit CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by foursixnine**
For the time being, only reference points where we have documentation, and point people to the right place

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -11,11 +11,17 @@ ask on our mailing list opensuse-factory@opensuse.org
...
-Please, find up-to-date documentation references on the official [openQA project web-page](http://open.qa/documentation/).
+Please, find up-to-date documentation references on the official [openQA project web-page](https://open.qa/documentation/).
 
 If you are looking for a task to start with, check out the [openQA Tests](https://progress.opensuse.org/projects/openqatests/issues/)
 Redmine project. Look for tickets with [easy] or [easy-hack] tags.
 
+#### Relevant documentation
+
+* All openQA documentation in a single [html page](https://open.qa/docs/)
+* openQA [test api](http://open.qa/api/testapi/) documentation
```
Comment (foursixnine) [inline CONTRIBUTING.md]:
```suggestion
* openQA [testapi](http://open.qa/api/testapi/) documentation
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -42,12 +47,18 @@ and additionally the following rules:
...
-  pull request. Call `make test` to conduct all tests.
 
+### Preparing a new Pull Request
+* All code needs to be tidy, for this use `make prepare` the first time you
+  set up your local environment, use `make tidy` or `tools/tidy` locally to
+  ensure your new code adheres to our coding style.
+* Every pull request is tested by the travis CI for different perl versions,
+  if something fails, run `make test` (don't forget to `make prepare` if your setup is new)
+  but the travis results are available too, in case they need to be investigated further
+* Whenever possible, [provide a verification run][1] of a job that runs the code [provided in the pull request][2]
```
Comment (okurz) [inline CONTRIBUTING.md]: Please break lines here at 78 characters

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -42,12 +47,18 @@ and additionally the following rules:
...
+
+Also see the [DoD/DoR][2] as a helpful (but not mandatory) guideline for new contributions.
+[1]: https://open.qa/docs/#_cloning_existing_jobs_openqa_clone_job
+[2]: https://progress.opensuse.org/projects/openqatests/wiki/Wiki#Definition-of-DONEREADY
+[3]: https://open.qa/docs/#_triggering_tests_based_on_an_any_remote_git_refspec_or_open_github_pull_request
```
Comment (okurz) [inline CONTRIBUTING.md]: Good idea to mention this but do you reference it in the text? Couldn't find it

File: CONTRIBUTING.md (hunk 2 above)
Comment (foursixnine) [reply CONTRIBUTING.md]: Why? There are lines in the same document that are longer than that. Looks rather ugly if changed :)... Terminals in 2019 handle more than 80 characters rather fine...

File: CONTRIBUTING.md (hunk 3 above)
Comment (foursixnine) [reply CONTRIBUTING.md]: Good catch! it's fixed now

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -42,12 +47,17 @@ and additionally the following rules:
...
+### Preparing a new Pull Request
+* All code needs to be tidy, for this use `make prepare` the first time you
+  set up your local environment, use `make tidy` or `tools/tidy` locally to
+  ensure your new code adheres to our coding style.
+* Every pull request is tested by the travis CI for different perl versions,
+  if something fails, run `make test` (don't forget to `make prepare` if your setup is new)
+  but the travis results are available too, in case they need to be investigated further
+* Whenever possible, [provide a verification run][1] of a job that runs the code [provided in the pull request][2]
+
+Also see the [DoD/DoR][2] as a helpful (but not mandatory) guideline for new contributions.
```
Comment (okurz) [inline CONTRIBUTING.md]: That needs to reference [3] now

Comment (foursixnine) [reply CONTRIBUTING.md]: fixed

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -11,11 +11,17 @@ ask on our mailing list opensuse-factory@opensuse.org
...
+Please, find up-to-date documentation references on the official [openQA project web-page](https://open.qa/documentation/).
 
 If you are looking for a task to start with, check out the [openQA Tests](https://progress.opensuse.org/projects/openqatests/issues/)
 Redmine project. Look for tickets with [easy] or [easy-hack] tags.
 
+#### Relevant documentation
+
+* All openQA documentation in a single [html page](https://open.qa/docs/)
+* openQA [testapi](http://open.qa/api/testapi/) documentation
+* (wip) Available test library documentation http://os-autoinst.github.io/openQA/
```
Comment (okurz) [inline CONTRIBUTING.md]: It's very likely that we either will forget to update this one or will never leave "wip" state. I suggest to simply delete the `(wip)` as it's the responsibility of a different repo to declare it "done" anyway.

Comment (foursixnine) [reply CONTRIBUTING.md]: I prefer to keep it, everything is WIP, but this in particular, still needs a lot of progress.

> Can you verify that it looks better when converted with asciidoctor?

We don't use asciidoctor for this repository. However there was a mistake in my last change: https://github.com/foursixnine/os-autoinst-distri-opensuse/blob/docs-update/CONTRIBUTING.md now it should render properly

**Comment by foursixnine** [comment]
> Can you verify that it looks better when converted with asciidoctor?

We don't use asciidoctor for this repository. However there was a mistake in my last change: https://github.com/foursixnine/os-autoinst-distri-opensuse/blob/docs-update/CONTRIBUTING.md now it should render properly

**Pull request merged**